### ISSUE TITLE
关于VAE 训练的问题

### ISSUE THREAD

**Issue by gogogogzb** [stepfun-ai/Step1X-3D#51]
作者您好，我们尝试对vae从头进行训练，发现了一些问题，不论是在我们通过您提供的watertight_and_sampling.py代码处理的大约3K个Objaverse数据，还是仅仅使用您在data/shape_autoencoder/objaverse/surfaces/000-000/00a1a602456f4eb188b522d7ef19e81b提供的一个样例上进行训练，vae都不能有效的收敛，甚至在前期训练过程中在验证阶段提取mesh时，还会报错（报错内容基本上是vae解码失败导致matching cubes算法无法提取网格表面）。
data_type: "Objaverse-datamodule"
data:
  root_dir: "./data/shape_autoencoder/objaverse"

  load_geometry: True                # whether to load geometry
  load_geometry_supervision: True    # whether to load geometry supervision
  geo_data_type: "sdf" 
  with_sharp_data: True             # original objaverse data
  n_samples: 32768
  noise_sigma: 0.
  random_flip: True # whether to randomly flip the input mesh
  random_color_jitter: True # whether to add random color jitter to the input images
  random_rotate: True # whether to add random rotation to the input images

  load_image: False                # whether to load images 
  load_caption: False              # whether to load captions 
  load_label: False                # whether to load labels

  batch_size: 10 # TODO: change this for your own dataset
  num_workers: 1 # change this for your own dataset

system_type: "shape-autoencoder-system"
system:
  sample_posterior: true
  bounds: 1.05
  mc_level: 0.0
  octree_resolution: 256

  shape_model_type: "michelangelo-autoencoder"
  shape_model:
    n_samples: ${data.n_samples}
    with_sharp_data: ${data.with_sharp_data} # dora config
    use_downsample: true
    num_latents: 1024
    embed_dim: 64
    point_feats: 3
    out_dim: 1 # tsdf(1)
    num_freqs: 8
    include_pi: false
    heads: 12
    width: 768
    num_encoder_layers: 8
    num_decoder_layers: 16
    use_ln_post: true
    init_scale: 0.25
    qkv_bias: false
    use_flash: true
    use_checkpoint: false

    volume_decoder_type: 'hierarchical'
    # volume_decoder_type: str = 'vanilla'
    surface_extractor_type: 'mc'

  loggers:
    wandb:
      enable: false
      project: "step1x-3d"
      name: shape-autoencoder+${name}+${tag}

  loss:
    lambda_kl: 0.001
    lambda_logits: 1.

  optimizer:
    name: AdamW
    args:
      lr: 1.e-5
      betas: [0.9, 0.99]
      eps: 1.e-6

trainer:
  num_nodes: 1
  max_epochs: 10
  log_every_n_steps: 5
  num_sanity_val_steps: 1
  check_val_every_n_epoch: 1
  enable_progress_bar: true
  precision: 32
  # precision: bf16-mixed

checkpoint:
  save_last: true
  save_top_k: -1
  every_n_train_steps: 5000
以上是我训练的参数，基本是按照您开源的yaml文件，不同的是 with_sharp_data这个，我看了一下Dora的论文，感觉应该是要使用sharp_surface_points的。
在大约3K处理过的Objaverse数据集上训练的loss曲线和验证时提取的mesh结果如下：

<img width="633" height="286" alt="Image" src="https://github.com/user-attachments/assets/fd1d7b46-7dfb-42b2-9eb1-d08aa9ab9957" />

<img width="378" height="234" alt="Image" src="https://github.com/user-attachments/assets/e2a16ce5-0706-4d05-bbdb-59645642acbe" />
这个结果是我随机选取的一个样例上解码的结果，很怪，大部分样例都是这个结果。
然后我又尝试在您提供的1个数据集上复制多份进行训练，loss曲线和mesh如下：

<img width="765" height="333" alt="Image" src="https://github.com/user-attachments/assets/6877fdfa-85eb-462a-8dde-691874d14c38" />

<img width="278" height="314" alt="Image" src="https://github.com/user-attachments/assets/daf2d6ed-10a3-43c5-a1ca-620e0450e34e" />

但是他真实的样子应该如图：

<img width="306" height="233" alt="Image" src="https://github.com/user-attachments/assets/47d64548-7959-4bba-a8c9-e96382987a9d" />
我很想知道您开源的michelangelo.yaml文件是您从头训练vae的配置文件吗，还有就是step1x3d_geometry/systems/shape_autoencoder.py这个代码是否是您训练时使用的代码？
